workers/data_refinery_workers/processors/test_compendia.py

```diff
@@ -591,44 +633,68 @@ def test_imputation(self):
             create_compendia.COMPENDIA_PIPELINE[:imputation_index],
         )
 
-        job_context["microarray_matrix"].loc[
-            "ENSDARG00000001313", "02c3v1urzlq9o499ye6315v3_GSM396530.PCL"
-        ] = np.nan
-
-        job_context["microarray_matrix"].loc[
-            "ENSDARG00000000241", "05y584nlulkpdchrjiwhjj7j_GSM448988.PCL"
-        ] = np.nan
+        # First, run the imputation step without removing anything to get a baseline
+        expected_context = utils.run_pipeline(
+            job_context.copy(), [create_compendia.COMPENDIA_PIPELINE[imputation_index]]
+        )
 
-        job_context["microarray_matrix"].loc[
-            "ENSDARG00000000151", "02c3v1urzlq9o499ye6315v3_GSM396530.PCL"
-        ] = np.nan
+        # Now pick some rows to remove according to the instructions from
+        # https://github.com/AlexsLemonade/refinebio/pull/2879#issuecomment-895143336
+
+        random.seed(42)
+
+        # Select some rows randomly and mask ~30% of the values
```

Comment:
Where does the 30% come in? Is that because there's like 366 columns?

Reply:
Good catch, turns out that when I rewrote the tests we ended up with a lot more columns than I expected.

I updated this code to be more explicit and fixed this to be the correct number.